api/android/api/src/main/java/org/nnsuite/nnstreamer/CustomFilter.java

```diff
@@ -65,20 +52,24 @@
     }
 
     /**
-     * Registers new custom-filter with name.
+     * Registers new custom-filter with input and output tensors information.
      *
+     * NNStreamer processes the tensors with 'custom-easy' framework which can execute without the model file.
      * Note that if given name is duplicated in the pipeline, the registration will be failed and throw an exception.
      *
      * @param name     The name of custom-filter
+     * @param in       The input tensors information
+     * @param out      Tne output tensors information
      * @param callback The function to be called while processing the pipeline
      *
      * @return {@link CustomFilter} instance
      *
      * @throws IllegalArgumentException if given param is null
      * @throws IllegalStateException if failed to initialize custom-filter
      */
-    public static CustomFilter registerCustomFilter(@NonNull String name, @NonNull CustomFilterCallback callback) {
-        return new CustomFilter(name, callback);
+    public static CustomFilter registerCustomFilter(@NonNull String name, @NonNull TensorsInfo in,
+            @NonNull TensorsInfo out, @NonNull CustomFilterCallback callback) {
```

Comment:
If Android API do not follow the backward compatibility, then it's OK.

Reply:
It should be ok for now as we do not have corresponding applications, yet.